### PR TITLE
GnuCash Help->GnuCash Manual and doc/lang->lang/doc directory structure

### PR DESCRIPTION
**Pull request by gjanssens**
* All build artefacts that were named _gnucash-help{.something}_ have been renamed to _gnucash-manual{.something}_
* Target names have been shortened. Wherever the target name contained _gnucash-help_ this has been replaced with _manual_. Similarly _gnucash-guide_ has been replaced with _guide_
* Source directories have been reshuffled. New structure is to have language directories at the top level. Each language directory can have one or more document directories (`guide|manual`).

This PR should be merged in sync with a related PR in the gnucash repository, the gnucash-on-macos repository and the gnucash-htdocs repository. The gnucash-on-windows and gnucash-on-flatpak repositories are not directly affected.

### CODE REVIEW

**Review by fellen — COMMENTED** [review]
We will also need to review [Documentation Improvement](https://wiki.gnucash.org/wiki/Documentation_Improvement) etc.

**Comment by gjanssens** [comment]
> We will also need to review [Documentation Improvement](https://wiki.gnucash.org/wiki/Documentation_Improvement) etc.

Yeah, but perhaps that's best done when we start the beta releases and at the same time discourage any further work on maint. Otherwise we have to keep two sets of documentation improvement guidelines.

File: README [hunk 1]
```diff
@@ -153,8 +153,8 @@ depending on the output format you require.
 You can also run xmllint on all documents or a specific document with the command
 
   make check # for toplevel check
-  make de-gnucash-help-check # for a cmake check for one specific document
-                             # in this example - the German help guide
+  make de-gnucash-manual-check # for a cmake check for one specific document
```
[the remark this answers is not in the document]
Comment (gjanssens) [reply README]: Good catch.

File: README [hunk 2]
```diff
@@ -141,7 +141,7 @@ languages at once by omitting the language specifier in the target.
 
 For example:
 
-  make gnucash-help-html
+  make gnucash-manual-html
```
[the remark this answers is not in the document]
Comment (gjanssens) [reply README]: Fixed as well.

**Comment by fellen** [comment]
> > We will also need to review [Documentation Improvement](https://wiki.gnucash.org/wiki/Documentation_Improvement) etc.
> 
> Yeah, but perhaps that's best done when we start the beta releases and at the same time discourage any further work on maint. Otherwise we have to keep two sets of documentation improvement guidelines.

You could already add it:
```
;From 4.900: <para 1>
: <para 2> …
;Up to 4.x, x<900: (Indent the old stuff by prepend paragraphs by a colon)

```
Then it does not get forgotten and current contributors learn about the difference.

**Pull request merged**